src/indexer/index.ts

```diff
@@ -183,45 +163,71 @@ export const isAssetOpted = async (
 /**
  * Check if given application is opted in given address
  * @param address Account address
- * @param application_id application id
+ * @param applicationId application id
  */
 export const isApplicationOpted = async (
 	address: string,
-	application_id: number
+	applicationId: number
 ): Promise<boolean> => {
 	try {
-		const optedApplicationInfo = await AlgoSigner.algod({
-			ledger: CHAIN_NAME,
-			path: `/v2/accounts/${address}/applications/${application_id}`,
-		});
-		if (optedApplicationInfo && optedApplicationInfo["app-local-state"]) {
-			return optedApplicationInfo["app-local-state"].id === application_id;
+		let isApplicationOpted = false;
+		const applicationInfo = await indexerClient
+			.lookupAccountAppLocalStates(address)
+			.do();
+		const parsedApplicationInfo = JSON.parse(JSON.stringify(applicationInfo));
+		if (parsedApplicationInfo && parsedApplicationInfo[LOCAL_STATE]) {
+			const optedApplicationInfo = applicationInfo[LOCAL_STATE].find(
+				(i: any) => (i[APPLICATION_ID] = applicationId)
+			);
+			if (optedApplicationInfo) {
+				isApplicationOpted = true;
+			}
 		}
-		return false;
+		return isApplicationOpted;
 	} catch (e) {
 		console.error(e);
 		throw e;
 	}
 };
 
-export const searchForGovASAToken = async () => {
-	const daoIdStore = DaoID();
-	const walletStore = WalletStore();
-	const govtID = DaoID().govt_id;
-	if (walletStore.address && govtID) {
-		try {
-			const accountInfo = await AlgoSigner.algod({
-				ledger: CHAIN_NAME,
-				path: `/v2/accounts/${walletStore.address}/assets/${govtID}`,
-			});
-			let amount = 0;
-			if (accountInfo["asset-holding"].amount) {
-				amount = accountInfo["asset-holding"].amount;
+export const getGovASATokenAmount = async (
+	address: string,
+	govtID: number
+): Promise<number> => {
+	try {
+		const assetInfo = await indexerClient.lookupAccountAssets(address).do();
+		let amount = 0;
+		if (assetInfo[ASSETS]) {
+			const govtAssetInfo = assetInfo[ASSETS].find(
+				(i: any) => i[ASSET_ID] === govtID
+			);
+			if (govtAssetInfo) {
+				amount = govtAssetInfo.amount;
 			}
-			daoIdStore.available = amount;
-		} catch (e) {
-			console.log(e);
-			throw e;
 		}
+		return amount;
+	} catch (e) {
+		console.log(e);
+		throw e;
 	}
 };
+
+export async function decodeStateMap(
+	state: Array<any>
+): Promise<Map<Key, StateValue>> {
+	const stateMap = new Map<Key, StateValue>();
+	if (state) {
+		for (const g of state) {
+			const key = Buffer.from(g.key, "base64").toString();
+			if (g.value.type === 1) {
```

Comment:
What is `1` meaning in this case?

Reply:
it signifies that it is bytes type not uint